x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/index.tsx

```diff
@@ -71,15 +69,6 @@ export const Transaction: React.SFC<Props> = ({
   location,
   waterfall
 }) => {
-  if (isEmpty(transaction)) {
```

Comment:
What was the bug here? It looks like we just moved this if check higher up?

Reply:
Yes, exactly. It needs to be moved higher up, else it will never be displayed.

`transaction` is passed to `WaterfallRequest`:
https://github.com/elastic/kibana/blob/24ba1908a3fff03ddba7d4023c66a0e79a52463c/x-pack/plugins/apm/public/components/app/TransactionDetails/view.tsx#L68-L78

If `transaction` is nil `WaterfallRequest` will return null, and not render any of its children (Like the `Transaction` component).
https://github.com/elastic/kibana/blob/24ba1908a3fff03ddba7d4023c66a0e79a52463c/x-pack/plugins/apm/public/store/reactReduxRequest/waterfallV1.tsx#L38